happi/cli.py

```diff
@@ -66,13 +67,15 @@ def happi_cli(ctx, path, verbose):
     logger.debug("Set logging level of %r to %r", shown_logger.name, level)
 
     # gather client
-    client = happi.client.Client.from_config(cfg=path)
-    logger.debug("Happi client: %r" % client)
-
-    # insert items into context to be passed to subcommands
-    # User objects must be assigned to ctx.obj, which will be passed
-    # through to new context objects
-    ctx.obj = client
+    try:
+        client = happi.client.Client.from_config(cfg=path)
+        logger.debug("Happi client: %r" % client)
+        # insert items into context to be passed to subcommands
+        # User objects must be assigned to ctx.obj, which will be passed
+        # through to new context objects
+        ctx.obj = client
```

Comment:
Is there any way we can give a better error message for the user when there is no client but they invoke a routine that requires one? I assume this one throws a cryptic `AttributeError`, though I haven't run it myself yet.

Reply:
My thought was initially to store the path in the `ctx.obj` attribute and call `Client.from_config` in each of the subcommands.  A bit more verbose but maybe less cryptic?

Reply:
This is why I was saying each command that requires a _valid and working_ client should instantiate the client on its own.

That is to say, `happi search` (and all others that require the client) should have a `get_cli_happi_client_from_config(ctx)`.

Were it structured like this, then exceptions could be raised or caught in a sensible spot - rather than before control flow gets to the command the user is trying to run.

Reply:
Do you want me to do this in this PR, or should we merge this and then do that refactor in a separate PR? I'm okay either way... really up to y'all.

Reply:
Do you mind doing that in this PR?  I think it should be a quick enough addition